include/exec/repeat_effect_until.hpp

```diff
@@ -0,0 +1,215 @@
+/*
+ * Copyright (c) 2023 Runner-2019
+ * Copyright (c) 2023 NVIDIA Corporation
+ *
+ * Licensed under the Apache License Version 2.0 with LLVM Exceptions
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *   https://llvm.org/LICENSE.txt
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#pragma once
+
+#include "../stdexec/execution.hpp"
+#include "exec/trampoline_scheduler.hpp"
+#include "exec/on.hpp"
+#include "__detail/__manual_lifetime.hpp"
+#include "stdexec/concepts.hpp"
+#include <concepts>
+
+namespace exec {
+  namespace __repeat_effect_until {
+    using namespace stdexec;
+
+    template <class _SourceId, class _ReceiverId>
+    struct __receiver {
+      struct __t;
+    };
+
+    template <class _SourceId, class _ReceiverId>
+    struct __operation {
+      using _Source = stdexec::__t<_SourceId>;
+      using _Receiver = stdexec::__t<_ReceiverId>;
+
+      struct __t : stdexec::__immovable {
+        using __id = __operation;
+        using __receiver_t = stdexec::__t<__receiver<_SourceId, _ReceiverId>>;
+
+        [[no_unique_address]] _Source __source_;
+        [[no_unique_address]] _Receiver __rcvr_;
+        trampoline_scheduler __sched_;
+
+        using __source_on_scheduler_sender = decltype(stdexec::on(__sched_, __source_));
```

Comment:
I'm not 100% sure this is doing what you want. Is `__source_` really an lvalue reference in this context? On all supported compilers? IMO it's better to be explicit with `__declval`.

Reply:
I try to switched to __call_result_t and used __declval